test/words/test/TestINodeWhile.java

```diff
@@ -0,0 +1,33 @@
+package words.test;
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import words.ast.*;
+import words.environment.*;
+import words.exceptions.WordsInvalidTypeException;
+import words.exceptions.WordsObjectAlreadyExistsException;
+import words.exceptions.WordsObjectNotFoundException;
+import words.exceptions.WordsRuntimeException;
+
+
+public class TestINodeWhile extends TestINode {
+	@Test
+	public void testWorkingWhile() throws WordsRuntimeException {
+		environment.createObject("Fred", "thing", new WordsPosition(0,0));
+		LNodeReference fredLNodeRef = new LNodeReference("Fred's");
+		INodeReferenceList fredRefList = new INodeReferenceList(fredLNodeRef);
+		
+		AST statementList = new INodeStatementList(new INodeAssign(fredRefList, new LNodeIdentifier("row"), new LNodeNum(environment.getObject("Fred").getCurrentCell().x++)));
+		AST relation = new INodeLess(new LNodeNum(environment.getObject("Fred").getCurrentCell().x), new LNodeNum(2));
+		AST testNode = new INodeWhile(relation, statementList);
+		
+		loop.enqueueAST(testNode);
```

Comment:
The one minor change I'd suggest is to replace lines 25:26 with a single line, which is testNode.eval().  Alternatively, you can leave it as is, but for clarity I would fastForwardEnvironment just 1, not 2, since the environment only needs to go through one frame to encounter and process the while loop.